### ISSUE TITLE
Use SQL for Internal Tag Database.

### ISSUE THREAD

**Issue by Penguin-Guru** [Bionus/imgbrd-grabber#2296]
I suspect this would speed things up but cannot make any promises. It would also be a convenient way for users to integrate other applications and manually import/export tags if they want to. Could be useful for backups and synchronisation.

**Comment by Bionus** [comment]
What do you mean by "tags" in this context?

There's already a SQLite tag database present but I guess it's not the same thing. The existing one is used assign a tag type (artist, copyright, character, etc.) to a given tag:
https://github.com/Bionus/imgbrd-grabber/issues/1033

**Comment by Penguin-Guru** [comment]
Yes, I meant to store as much of the source metadata as possible so it can be archived and queried locally.

It would probably make sense to provide a setting that allows users to control which media's metadata is added, such as all media viewed, all saved search results, and all favourited media.
